packages/riverpod_lint/lib/src/assists/convert_to_stateless_base_widget.dart

```diff
@@ -0,0 +1,241 @@
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/source/source_range.dart';
+// ignore: implementation_imports, somehow not exported by analyzer
+import 'package:analyzer/src/generated/source.dart' show Source;
+import 'package:collection/collection.dart';
+import 'package:custom_lint_builder/custom_lint_builder.dart';
+
+import '../riverpod_custom_lint.dart';
+import 'convert_to_widget_utils.dart';
+
+class ConvertToStatelessBaseWidget extends RiverpodAssist {
+  ConvertToStatelessBaseWidget({
+    required this.targetWidget,
+  });
+  final StatelessBaseWidgetType targetWidget;
+  late final statelessBaseType = getStatelessBaseType(
+    exclude: targetWidget,
+  );
+  late final statefulBaseType = getStatefulBaseType(
+    exclude: targetWidget == StatelessBaseWidgetType.statelessWidget
+        ? StatefulBaseWidgetType.statefulWidget
+        : null,
+  );
+
+  @override
+  void run(
+    CustomLintResolver resolver,
+    ChangeReporter reporter,
+    CustomLintContext context,
+    SourceRange target,
+  ) {
+    if (targetWidget.requiredPackage != null &&
+        !context.pubspec.dependencies.keys
+            .contains(targetWidget.requiredPackage)) {
+      return;
+    }
+
+    context.registry.addExtendsClause((node) {
+      // Only offer the assist if hovering the extended type
+      if (!node.superclass.sourceRange.intersects(target)) return;
+
+      final type = node.superclass.type;
+      if (type == null) return;
+
+      if (statelessBaseType.isExactlyType(type)) {
+        _convertStatelessToStatelessWidget(
+          reporter,
+          node,
+        );
+        return;
+      }
+
+      if (statefulBaseType.isExactlyType(type)) {
+        final isExactlyStatefulWidget = StatefulBaseWidgetType
+            .statefulWidget.typeChecker
+            .isExactlyType(type);
+
+        _convertStatefulToStatelessWidget(
+          reporter,
+          node,
+          resolver.source,
+          // This adjustment assumes that the priority of the standard "Convert to StatelessWidget" is 30.
+          isExactlyStatefulWidget ? -4 : 0,
```

Comment:
Use named parameters

Reply:
Also, what is this supposed to be doing?

Reply:
This adjustment is due to the fact that the existing Convert to StatefulWidget and Convert to StatelessWidget both have a priority of 30.
https://github.com/dart-lang/sdk/blob/9b1798e08136b3d29d929fb120f3bb2e1492de09/pkg/analysis_server/lib/src/services/correction/assist.dart#L224-L233

The priority of assists is generally set with Convert to StatefulWidget as 30, but the order will not be as expected when the source Widget is a StatefulWidget.

![SCR-20230408-st6](https://user-images.githubusercontent.com/90010509/230719369-de05094f-9f6e-4692-b0c9-3c01e6f04d9d.png)

By making an adjustment of -4 to the priority only when it's a StatefulWidget, the priority is based on Convert to StatelessWidget 30, resulting in the expected display.

Reply:
I see, thanks!